File: src/main/java-graphql/io/github/graphql/model/PinnableItemResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for PinnableItem
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:48+0800"
)
public class PinnableItemResponseProjection extends GraphQLResponseProjection {

    public PinnableItemResponseProjection() {
    }

    @Override
    public PinnableItemResponseProjection all$() {
        return all$(3);
    }

    @Override
    public PinnableItemResponseProjection all$(int maxDepth) {
        this.typename();
        return this;
    }

    public PinnableItemResponseProjection onGist(GistResponseProjection subProjection) {
        return onGist(null, subProjection);
    }

    public PinnableItemResponseProjection onGist(String alias, GistResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("...on Gist").alias(alias).projection(subProjection));
        return this;
    }

    public PinnableItemResponseProjection onRepository(RepositoryResponseProjection subProjection) {
        return onRepository(null, subProjection);
    }

    public PinnableItemResponseProjection onRepository(String alias, RepositoryResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("...on Repository").alias(alias).projection(subProjection));
        return this;
    }

    public PinnableItemResponseProjection typename() {
        return typename(null);
    }

    public PinnableItemResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/graphql/io/github/graphql/model/PinnableItemResponseProjection.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for PinnableItem
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-05T19:09:00+0800"
)
open class PinnableItemResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): PinnableItemResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): PinnableItemResponseProjection {
        this.typename()
        return this
    }

    fun onGist(subProjection: GistResponseProjection): PinnableItemResponseProjection = onGist(null, subProjection)

    fun onGist(alias: String?, subProjection: GistResponseProjection): PinnableItemResponseProjection {
        fields.add(GraphQLResponseField("...on Gist").alias(alias).projection(subProjection))
        return this
    }

    fun onRepository(subProjection: RepositoryResponseProjection): PinnableItemResponseProjection = onRepository(null, subProjection)

    fun onRepository(alias: String?, subProjection: RepositoryResponseProjection): PinnableItemResponseProjection {
        fields.add(GraphQLResponseField("...on Repository").alias(alias).projection(subProjection))
        return this
    }

    fun typename(): PinnableItemResponseProjection = typename(null)

    fun typename(alias: String?): PinnableItemResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }


}
